packages/client/tests/functional/logging-mongo/tests.ts

```diff
@@ -0,0 +1,73 @@
+import testMatrix from './_matrix'
+
+// @ts-ignore this is just for type checks
+declare let prisma: import('@prisma/client').PrismaClient
+
+// @ts-ignore this is just for type checks
+type PrismaClient = import('@prisma/client').PrismaClient
+// @ts-ignore this is just for type checks
+declare let newPrismaClient: NewPrismaClient<PrismaClient>
+
+testMatrix.setupTestSuite(
+  () => {
+    test('basic event logging', async () => {
+      const prisma = newPrismaClient({
+        log: [
+          {
+            emit: 'event',
+            level: 'query',
+          },
+        ],
+      })
+
+      const onQuery = jest.fn()
+
+      prisma.$on('query', onQuery)
+
+      await prisma.user.findMany()
+
+      expect(onQuery).toHaveBeenCalledWith({
+        params: expect.any(String),
+        query: expect.any(String),
+        target: 'mongodb_query_connector::query',
+        timestamp: expect.any(Date),
+      })
+    })
+
+    test('interactive transactions logging', async () => {
+      const prisma = newPrismaClient({
+        log: [
+          {
+            emit: 'event',
+            level: 'query',
+          },
+        ],
+      })
+
+      const onQuery = jest.fn()
+
+      prisma.$on('query', onQuery)
+
+      await prisma.$transaction(async (tx) => {
+        await tx.user.findMany()
+      })
+
+      expect(onQuery.mock.calls).toEqual([
+        [
+          {
+            params: expect.any(String),
+            query: expect.any(String),
+            target: 'mongodb_query_connector::query',
+            timestamp: expect.any(Date),
+          },
+        ],
+      ])
+    })
+  },
+  {
+    optOut: {
+      from: ['sqlite', 'postgresql', 'mysql', 'cockroachdb', 'sqlserver'],
+      reason: 'SQL databases tested in logging suite',
```

Comment:
There isn't much difference between the two tests, wondering if we could merge all the providers into one and not have the separate mongo logging, wdyt?

Reply:
I've started with a single suite, but branched mongo ones into separate one. There are much more differences then might seem from a first glance:

- mongo does not start connection pool and has no message about it
- `target` field for all events has different value for mongo,
- `duration` field is missing
- "begin transaction" and "end transaction" messages are missing.

Before the spit, every assert looked like "if it's mongo check if it's one thing, if it isn't check the different one"